SmartVision/WindowsForms/WebcamInput.cs

```diff
@@ -107,8 +108,17 @@ private static async void ProcessFrameAsync()
             while (await buffer.OutputAvailableAsync())
             {
                 byte[] frameToProcess = await buffer.ReceiveAsync();
-                var result = FaceRecognition.AnalyzeImage(frameToProcess);
-                Debug.WriteLine(DateTime.Now + " " + result);
+                if (buffer.Count >= BUFFER_LIMIT * 0.9)
+                {
+                    Debug.WriteLine("Buffer size within 10% of limit, skipping processing.");
+                    Debug.WriteLine(buffer.Count.ToString() + ">=" + (BUFFER_LIMIT * 0.9).ToString());
```

Comment:
Since the buffer is FIFO, each frame received for analysis is the oldest one at that point in time. 
If the buffer is nearly full, this frame is simply not sent for analysis. No normal operation is interrupted, the while loop still continues as long as there's frames in the buffer. Several frames might be skipped at a time if processing is slow enough.

Reply:
Here' an excerpt from the debug log, given that 90% of limit is 5:
Buffer size within 10% of limit, skipping processing.
6>=5
Buffer size within 10% of limit, skipping processing.
5>=5
Starting processing of frame. Remaining frames: 4

Reply:
Oh, now I get it. Thanks for the explanation. Great job!